internal/cumulative/cumulative.go

```diff
@@ -0,0 +1,138 @@
+package cumulative
+
+import (
+	"bytes"
+	"fmt"
+	pprofile "github.com/google/pprof/profile"
+	"github.com/pyroscope-io/client/upstream"
+	"time"
+)
+
+type ProfileMerger struct {
+	SampleTypes      []string
+	MergeRatios      []float64
+	SampleTypeConfig map[string]*upstream.SampleType
+
+	prev *pprofile.Profile
+}
+
+type Mergers struct {
+	Heap  *ProfileMerger
+	Block *ProfileMerger
+	Mutex *ProfileMerger
+}
+
+func NewMergers() *Mergers {
+	return &Mergers{
+		Block: &ProfileMerger{
+			SampleTypes: []string{"contentions", "delay"},
+			MergeRatios: []float64{-1, -1},
+			SampleTypeConfig: map[string]*upstream.SampleType{
+				"contentions": {
+					DisplayName: "block_count",
+					Units:       "lock_samples",
+				},
+				"delay": {
+					DisplayName: "block_duration",
+					Units:       "lock_nanoseconds",
+				},
+			},
+		},
+		Mutex: &ProfileMerger{
+			SampleTypes: []string{"contentions", "delay"},
+			MergeRatios: []float64{-1, -1},
+			SampleTypeConfig: map[string]*upstream.SampleType{
+				"contentions": {
+					DisplayName: "mutex_count",
+					Units:       "lock_samples",
+				},
+				"delay": {
+					DisplayName: "mutex_duration",
+					Units:       "lock_nanoseconds",
+				},
+			},
+		},
+		Heap: &ProfileMerger{
+			SampleTypes: []string{"alloc_objects", "alloc_space", "inuse_objects", "inuse_space"},
+			MergeRatios: []float64{-1, -1, 0, 0},
+			SampleTypeConfig: map[string]*upstream.SampleType{
+				"alloc_objects": {
+					Units: "objects",
+				},
+				"alloc_space": {
+					Units: "bytes",
+				},
+				"inuse_space": {
+					Units:       "bytes",
+					Aggregation: "average",
+				},
+				"inuse_objects": {
+					Units:       "objects",
+					Aggregation: "average",
+				},
+			},
+		},
+	}
+}
+
+// todo should we disableClientSideMerge if we fail to merge or keep trying?
+// todo should we filter by enabled ps.profileTypes to reduce profile size ? maybe add a separate option ?
```

Comment:
Now that we parse profile types we can filter profileTypes more honestly, for example we can send only "alloc_objects"

Reply:
That would be perfect: https://github.com/pyroscope-io/client/issues/13